pkg/app/api/grpcapi/web_api.go

```diff
@@ -1501,3 +1502,106 @@ func (a *WebAPI) getInsightDataForChangeFailureRate(
 		Value:     changeFailureRate,
 	}, nil
 }
+
+// getInsightDataForMTTR accumulate insight data in target range for mean time to restore
+func (a *WebAPI) getInsightDataForMTTR(
+	ctx context.Context,
+	projectID string,
+	applicationID string,
+	targetRangeFrom time.Time,
+	targetRangeTo time.Time) (*model.InsightDataPoint, error) {
+
+	var applicationIDs []string
+	if applicationID == "" {
+		apps, err := a.applicationStore.ListApplications(ctx, datastore.ListOptions{})
+		if err != nil {
+			a.logger.Error("failed to get applications", zap.Error(err))
+			return nil, status.Error(codes.Internal, "Failed to get applications")
+		}
+
+		for _, app := range apps {
+			applicationIDs = append(applicationIDs, app.Id)
+		}
+	} else {
+		applicationIDs = append(applicationIDs, applicationID)
+	}
+
+	var mttrs []float32
+	for _, appID := range applicationIDs {
+		filters := []datastore.ListFilter{
+			{
+				Field:    "ProjectId",
+				Operator: "==",
+				Value:    projectID,
+			},
+			{
+				Field:    "CreatedAt",
+				Operator: ">=",
+				Value:    targetRangeFrom.Unix(),
+			},
+			{
+				Field:    "CreatedAt",
+				Operator: "<",
+				Value:    targetRangeTo.Unix(), // target's finish time on unix time
+			},
+			{
+				Field:    "ApplicationId",
+				Operator: "==",
```

Comment:
Actually, we can use the "in" operator. With it, you can query a specific field for multiple values in a single query. (Maybe we should clearly define what operator is available.

Reply:
OK, fix to use `in` operator, thanks.